### PR TITLE
[Refactor] Network upgrades management cleanup

### PR DESCRIPTION
**Pull request by random-zebra**
Based on top of:
- [x] #1665 

Refactor the past height-based network upgrades with the new interface provided by #1665

### CODE REVIEW

**Comment by random-zebra** [comment]
Rebased on master, now that #1665 has been merged.

**Comment by random-zebra** [comment]
Rebased due to conflicts with #1670 just merged

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -738,27 +738,30 @@ UniValue verifychain(const UniValue& params, bool fHelp)
...
     case 7:
-        activated = pindex->nHeight >= consensusParams.height_start_TimeProtoV2;
+        idx = Consensus::UPGRADE_V4_0;
         break;
+    default:
+        rv.push_back(Pair("status", false));
+        return rv;
     }
-    rv.push_back(Pair("status", activated));
+    rv.push_back(Pair("status", consensusParams.NetworkUpgradeActive(pindex->nHeight, idx)));
```
Comment (furszy) [inline src/rpc/blockchain.cpp:764]: We can remove all of the hard forks from this soft fork description method (with the exception of bip65). Are going to appear twice in the `getblockchaininfo` RPC command as the "upgrades" description method call (`NetworkUpgradeDescPushBack` method call) is containing them too.

Comment (random-zebra) [reply src/rpc/blockchain.cpp:764]: Only bip65 will appear twice as `SoftForkDesc` is only ever called for it. But agree we can clean up this method in a successive PR.

**Comment by random-zebra** [comment]
Merging...

**Pull request merged**